miniupnpc/upnpc.c

```diff
@@ -705,15 +709,19 @@ int main(int argc, char ** argv)
 				break;
 			case 2:
 				printf("Found a (not connected?) IGD : %s\n", urls.controlURL);
+				if (!ignore) goto invalid;
```

Comment:
I don't like the goto very much ;)

Reply:
Ok, do you have any suggestion or improvement, how to achieve this change without goto?

Reply:
it also introduces a memory leak

Reply:
Now I see it... I will think a bit more about this code.

Reply:
see 1a1dc0f